### PR TITLE
[DTM] SOFT-4207 [12/15]: wait for a web font before rendering in it

### PR DESCRIPTION
**Pull request by d4mation**
🎫  https://linear.app/nexcess/issue/SOFT-4207/font-families-become-favorites-not-tokens
:video_camera:  https://www.loom.com/share/2cde91916171494694080f3b7beda6e9

Picking a font flashes: the browser paints the fallback face immediately and swaps when the file lands (`display=swap`), so a first pick of any family shows the wrong font for a moment. Loading earlier only moves the cost around. Keeping the current font on screen until the new face is usable, then switching once, is what removes it — at **no extra requests**, since it is the same single file.

Two waits, and the order is the whole point. `fonts.load()` resolves against the faces a document currently knows, so asking right after injecting a `<link>` resolves against none and reports success while the font is still missing. The link's own load event is what says the `@font-face` rules exist; only then is there anything to fetch.

Every wait is bounded. A caller holds the user's pick until this settles, so it always settles — offline, a blocked request, or a family Google does not serve ends in the font not being there, never in a promise that never resolves.

Takes the target document rather than assuming one: the editor canvas is an iframe with its own font set, and a stylesheet in the outer document styles nothing the user can see. Nothing here knows which host is asking.

Nothing consumes it yet.

### Checklist
- [x] I have performed a self-review.
- [x] No unrelated files are modified.
- [x] No debugging statements exist (Ex: console.log, error_log).
- [x] There are no warnings or notices in the wordpress error log.
- [x] Passes all tests (linting, acceptance, & unit)

### Block specific checklist (where relevant)
- [ ] Tested with an existing instance of this block .
- [ ] Tested creating a new instance of this block.
- [ ] Tested with Dynamic content & Elements.

## Summary by CodeRabbit

* **New Features**
  * Added font-loading support that waits for stylesheets and font faces to be ready before rendering.
  * Added Google Fonts URL generation and shared stylesheet loading.
  * Added support for document-specific contexts, including iframes.
  * Added bounded handling for failed or delayed font loads without blocking rendering.

* **Tests**
  * Added comprehensive coverage for font loading, deduplication, timeouts, failures, and edge cases.

### CODE REVIEW

File: src/token-controls/helpers/font-loading.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+			link.onload = resolve;
+			link.onerror = resolve;
+			doc.head.appendChild(link);
+		}),
+		timeout
+	);
+
+	sheets.set(href, ready);
+
+	return ready;
```
[the remark this answers is not in the document]
Comment (d4mation) [reply src/token-controls/helpers/font-loading.js]: Addressed in 415b5d139

**Review by pauloiankoski — APPROVED** [review]
